Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1367,6 +1394,51 @@ bool RenderBlockFlow::shouldTrimChildMargin(MarginTrimType marginTrimType, const
     return lastInFlowChildBox() == &child;
 }
 
+void RenderBlockFlow::trimFloatBlockEndMargins(LayoutUnit blockFormattingContextInFlowContentHeight)
+{
+    ASSERT(m_floatingObjects && createsBlockFormattingContext());
+    auto computeFloatMarginAfterStartLocationInBlockFormattingContext = [&](const RenderBox& floatBox) {
+        // Start with the location of the float within its containing block. If this containing
+        // block is the BFC in which we are trimming, then we can use this as the starting location
+        // since this is the float's positin within the BFC. If not, add the containing block's location
+        // to the float's location. This new value should be the location of the float relative to this
+        // new containing block. Keep doing this until we find the containing block that establishes
+        // a BFC
+        auto startLocation = logicalTopForChild(floatBox);
+        auto containingBlock = floatBox.containingBlock();
+        ASSERT(containingBlock);
+        while (!containingBlock->createsBlockFormattingContext()) {
+            startLocation += logicalTopForChild(*containingBlock);
+            containingBlock = containingBlock->containingBlock();
+        }
```

Comment:
don't we have a helper on this (translate some coords relative to an ancestor)?

Reply:
Also please add deeply nested type of test cases, where the float is a descendent of block boxes with both layout and paint type of offsets (e.g. border, padding and relatively positioned, transformed etc).

Reply:
Added a few of those tests. Also after talking with Simon it seems like using `localToAbsolute` is reasonable to use while walking up the containing block chain here